packages/proxy/src/proxy.ts

```diff
@@ -187,11 +198,29 @@ export async function proxyV1({
   }
 
   // Caching is enabled by default, but let the user disable it
-  const useCacheMode = parseEnumHeader(
+  // XXX is there a clean way to specify "auto" caching behavior using only cache-control directives?
+  const legacyCacheMode = parseEnumHeader(
     CACHE_HEADER,
     CACHE_MODES,
     proxyHeaders[CACHE_HEADER],
   );
+
+  let useCacheMode = legacyCacheMode;
+  const cacheControlDirectives = cacheControlParse(
+    proxyHeaders[CACHE_CONTROL_HEADER] || "",
+  );
+  // XXX does it make sense to have separate cache control for read and write?
```

Comment:
i think that does make sense, if you'd like to do that in this change

Reply:
I guess the only behavior that i would ever want to use that we don't support is forcibly "busting" the cache, where we don't read the current cache entry if any, but always write--up to you on whether we want to support that, we could also just keep this simple for now